### ISSUE TITLE
Drives not terminating automatically

### ISSUE THREAD

**Issue by chinedu40** [adriankumpf/teslamate#1731]
**Describe the bug**

Using my current setup, while drives are logged, they need to be manually terminated using the terminal in order to show up in the drives and trips sections.  i use the follow script to do so:

#!/bin/bash
for i in {..}
do
docker-compose exec teslamate bin/teslamate rpc \
    "TeslaMate.Repo.get!(TeslaMate.Log.Drive, $i) |> TeslaMate.Log.close_drive()"
done
exit 0

If they are not manually terminated, they is no record of trip taking place in the drives section, however, there there is a record in the trips section, but with no details. 

**Expected behavior**

I expect for trips to be automatically terminated and appear fully detailed in the drives and trips section. 

**How to reproduce it (as minimally and precisely as possible)**:

I am not sure how reproducible it is, but it appears similar to the bug described here #926 

**Relevant entries from the logs**

```
An example of a drive:

![Screenshot 2021-06-17 17 18 12](https://user-images.githubusercontent.com/6845955/122435715-24870200-cf90-11eb-9841-db8bc6d9129c.png)
![Screenshot 2021-06-17 17 18 33](https://user-images.githubusercontent.com/6845955/122435740-294bb600-cf90-11eb-88a8-46176686f558.png)
![Screenshot 2021-06-17 17 18 51](https://user-images.githubusercontent.com/6845955/122435760-2d77d380-cf90-11eb-8045-75b4fc83ff7b.png)




**Environment**

- TeslaMate version: 1.23.3
- Type of installation: Docker 
- OS TeslaMate is installed on: Synology DSM 6.2
- User OS & Browser: Chromebook OS

**Comment by chinedu40** [comment]
![Screenshot 2021-06-17 17 18 51](https://user-images.githubusercontent.com/6845955/122436224-8e9fa700-cf90-11eb-9708-709d9c823229.png)
![Screenshot 2021-06-17 17 18 33](https://user-images.githubusercontent.com/6845955/122436231-8fd0d400-cf90-11eb-9261-d688d2badd0a.png)
![Screenshot 2021-06-17 17 18 12](https://user-images.githubusercontent.com/6845955/122436233-8fd0d400-cf90-11eb-96c2-be18d3e5c623.png)

**Comment by adriankumpf** [comment]
The error messages in your screenshot provide the crucial hint:

```
DBConnection.ConnectionError: (DBConnection.ConnectionError) tcp recv: closed (the connection was closed by the pool, possibly due to a timeout or because the pool has been terminated)
```

This database error indicates an issue with the connection to the database. It could be that the database can only process requests very slowly because your hardware is overloaded at that time or is busy with other things that are running in the background.

It' s not a TeslaMate bug, because it is actually trying to terminate the drive properly as shown in the screenshot. However, this operation fails because the database is not responding.

**Comment by chinedu40** [comment]
Thanks for the explanation. I'm running it on Synology NAS via docker. I've upped the CPU priority to the highest and so far drives are terminating automatically and showing up in the drives section.

**Comment by adriankumpf** [comment]
All right 👍Then I'll close this thread for now.

**Comment by w0rd3r** [comment]
> Thanks for the explanation. I'm running it on Synology NAS via docker. I've upped the CPU priority to the highest and so far drives are terminating automatically and showing up in the drives section.

Hello, I have the same issue since couple of days ago, drives not ending. 
I have this in the logs:

`2021-07-10 06:59:22.183 [error] Postgrex.Protocol (#PID<0.2281.0>) disconnected: ** (DBConnection.ConnectionError) client #PID<0.2527.0> timed out because it queued and checked out the connection for longer than 60000ms`

I changed environment variable DATABASE_TIMEOUT to 120 000 instead of 60 000 and it solved the issue for now. 

Could I ask if you think the database cannot handle too many drives @adriankumpf ? I can't think of a reason why this appeared this week. I also found out that when I stop a drive and park the car, Teslamate becomes unresponsive for about 30 seconds. 

See screenshot attached. 1 drive; 2 just after park; 3 after about 30 seconds in park 

![B thumb jpg e4d85c0252e801b706516a6f0d2e1774](https://user-images.githubusercontent.com/10288558/125176973-72320b80-e1d8-11eb-8e93-067908cca380.jpg)
